public/scripts/stocklist.coffee

```diff
@@ -0,0 +1,115 @@
+R = React.DOM
+
+AddProduct = React.createClass
+  handleSubmit: (e) ->
+    e.preventDefault()
+    cat = (React.findDOMNode @refs.category).value.trim()
+    name = (React.findDOMNode @refs.name).value.trim()
+    price = (React.findDOMNode @refs.price).value
+    stock = (React.findDOMNode @refs.stocked).checked
+    if not cat or not name or not price
+      return
+    @props.addProduct {category: cat, price: price, stocked: stock, name: name}
+    (React.findDOMNode @refs.category).value = ''
+    (React.findDOMNode @refs.name).value = ''
+    (React.findDOMNode @refs.price).value = ''
+    (React.findDOMNode @refs.stocked).checked = false
+    return
+  render: ->
```

Comment:
this will get a lot easier to read if you have a line like:
`{div, option, select, h3, form, input, text} = R`
that way you dont have to do `R.<x>` everywhere